src/query/plan/variable_start_planner.cpp

```diff
@@ -209,19 +221,43 @@ CartesianProduct<VaryMatchingStart> VaryMultiMatchingStarts(const std::vector<Ma
   return MakeCartesianProduct(std::move(variants));
 }
 
+CartesianProduct<VaryMatchingStart> VaryFilterMatchingStarts(const Matching &matching,
+                                                             const SymbolTable &symbol_table) {
+  auto i = 0;
+  for (const auto &filter : matching.filters) {
+    for ([[maybe_unused]] const auto &filter_matchings : filter.matchings) {
```

Comment:
why is this for loop?

can't we change that to:

if (filter.matching){
    i+=1;
}

Reply:
there can be multiple. best i can do is size